### DISCUSSION TITLE
help on lua script for custom writer

### DISCUSSION THREAD

**Discussion by AllgemeineVerunsicherung** [jgm/pandoc#9427] [Q&A]
I'd like to specify a class to HTML blockquote. I created the following Lua script:

```lua
Writer = pandoc.scaffolding.Writer

function Writer ( doc, opts )
  local filter = {
    BlockQuote = function( bq )
      local fb = '<blockquote class="foobar">\n'.. '' ..'\n</blockquote>'
      return pandoc.RawBlock('html5', fb)
    end
  }
  return pandoc.write(doc:walk(filter), 'html5', opts)
end
```

The part I am missing is the content of the blockquote. I tried with `bq.text` and `bq.content` but it gives me error:

`attempt to concatenate a nil value (field 'text')`

Blockquote needs the following structure from AST:  `BlockQuote [ Para [ Str "a" , Space , Str "citation" ] ]`

**Comment by fiapps** [comment]
`bq.content` or `bq.c`will work. You want something like this:

```lua
Writer = pandoc.scaffolding.Writer

function Writer ( doc, opts )
  local filter = {
    BlockQuote = function( bq )
      bq.content:insert(1, pandoc.RawBlock('html5', '<blockquote class="foobar">'))
      bq.content:insert(pandoc.RawBlock('html5', '</blockquote>'))
      return bq.content
    end
  }
  return pandoc.write(doc:walk(filter), 'html5', opts)
end
```
This inserts the raw HTML you want at the beginning and end of the content list, and then returns the modified content list as a replacement for the BlockQuote element.
